### PR TITLE
Removing TSLint and replacing with ESLint

### PR DESCRIPTION
**Pull request by RyanChristian4427**
As the title suggests, I removed TSLint and replaced it with ESLint, as TSLint is now deprecated. While there is a tool to translate TSLint configurations to ESLint, that created a rather large mess, and still required TSLint to be installed, so I just decided to clear out the rules, besides the bare minimum. If you want, I can do the conversion, but I don't think it makes a ton of sense, personally.

In addition, I updated many of the dependencies, cleaned up/separated a few of the configurations, and ran the new linter through the project.

### CODE REVIEW

File: template/src/components/header/index.tsx [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import { h } from "preact";
+import preact, { h } from "preact";
```
Comment (ForsakenHarmony) [inline template/src/components/header/index.tsx]: what's that about?

Comment (RyanChristian4427) [reply template/src/components/header/index.tsx]: Just the IDE fixing a poor typing that I made. Will fix.

File: template/package.json [hunk 2]
```diff
@@ -10,90 +10,55 @@
...
-        "tslint-config-prettier": "^1.17.0",
-        "tslint-consistent-codestyle": "^1.14.1",
-        "tslint-eslint-rules": "^5.4.0",
-        "tslint-react": "^3.6.0",
-        "typescript": "^3.2.1",
-        "typings-for-css-modules-loader": "^1.7.0",
-        "css-loader": "^1.0.1"
+    "peerDependencies": {
+        "prettier": ">1.0.0",
+        "eslint": ">6.0.0"
```
Comment (ForsakenHarmony) [inline template/package.json]: do these have any value? I think we could remove them, not sure why we had them before?

Comment (RyanChristian4427) [reply template/package.json]: I don't think they do. Couldn't see how they'd be useful in a template. I'll have them removed.

**Comment by ForsakenHarmony** [comment]
Thanks for your contribution :tada:

**Pull request merged**